src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -3771,6 +3780,14 @@ namespace EvaporativeCoolers {
                     ZoneEvapUnit(UnitLoop).ActualFanVolFlowRate = HVACFan::fanObjs[ZoneEvapUnit(UnitLoop).FanIndex]->designAirVolFlowRate;
                     ZoneEvapUnit(UnitLoop).FanAvailSchedPtr = HVACFan::fanObjs[ZoneEvapUnit(UnitLoop).FanIndex]->availSchedIndex;
                 }
+
+                if (ZoneEvapUnit(UnitLoop).FanType_Num == DataHVACGlobals::FanType_SimpleOnOff ||
+                    ZoneEvapUnit(UnitLoop).FanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
+                    ZoneEvapUnit(UnitLoop).OpMode = DataHVACGlobals::CycFanCycCoil;
+                } else {
+                    ZoneEvapUnit(UnitLoop).OpMode = DataHVACGlobals::ContFanCycCoil;
+                }
+
```

Comment:
This is new code added in Get Input to catch and set the fan operating mode.  Later the "OpMode" is used whether to allow or not allow the unit to cycle.

Reply:
Oh I see, onoff is triggered by fan type. This won't fly after we deprecate all of the old fans and move toFan:SystemModel as the only fan. This unit probably needs a Supply Fan Operating Mode Schedule Name, similar to PTAC, PTHP, FanCoil, etc.

Reply:
@mjwitte are you requesting he add this on this PR?  If not, are you satisfied with the current state?

Reply:
I think we should add the field now. It does require some thought for the variable fan case. Does it currently check the settings inside Fan:SystemModel to determine if it's VAV?

Reply:
OK. I will append new field at the end. But we do not want to stretch this issue into "new feature".

Reply:
Before you code it, let's think through all the use cases and make sure it's clean. e.g. If someone currently has VAV, should this schedule be always 1?

Reply:
Here are the different option to consider:

(1) if the fan is VAV (Fan:VariableVolume or Fan:SystemModel with continuous Speed Control Method) then the schedule should be always 1 (no cycling) regardless of the fan op schedule. This requires to scan the Fan:SystemModel object. I do not know if there a method to tell or how easy it is to find out.

(2) Fan:OnOff and Fan:SystemModel (with number of speed = 1 with continuous "Speed Control Method:) should default to cycling fan unless the FanOpschedule tells us differently

(3)  Fan:ConstantVolume defaults to continuous fan always.

(4) What should we do if the input field is blank? The default operating mode should be dependent on the fan type, shouldn't it?

Reply:
<sigh> nothing simple. Question is - how important is it to reproduce past behavior? If the field is blank, then it should default to cycling fan, regardless of fan type.
What does, say, PTAC do if you specify Fan:ConstantVolume with a zero fan operating mode schedule? Does it reject that, or go ahead and cycle the Fan:ConstantVolume?

Reply:
If you specify Fan:ConstantVolume with a zero fan operating mode schedule value, then EPlus throws severe error message and fatal out. See code [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/PackagedTerminalHeatPump.cc#L1381-L1406). This seems very restrictive. On the other hand if you instead leave the supply air fan operating schedule name field blank, then EPlus defaults to cycling mode and simulation continues. See the code [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/PackagedTerminalHeatPump.cc#L1304-L1311).

Reply:
Current code allows zone Evap cooler to run in continuously mode regardless of the fan type supported.

Reply:
If we implement with approach used in PTAC and the supply air fan operation schedule is left bank, then the Zone Evap Cooler cycles regardless of the fan type for ZoneTemperatureDeadbandOnOffCycling and ZoneCoolingLoadOnOffCycling Unit Control Methods but has no effect when the Unit Control Method is ZoneCoolingLoadVariableSpeedFan.

But we can always reproduce the the past behavior if set the supply air fan operation schedule value is set to 1.

Reply:
@mjwitte OK. I will proceed with Fan Op Mode implementation similar to PTAC but with possibility of providing transition rule to reproduce past results. Let me know what you think?

Reply:
@MJW @Myoldmopar Second look at allowing the fan operating mode for zone Evap Unit: 

The existing implementation is not a "Continuous Fan Flow", rather it is FULL fan flow if the Evap Unit is ON based on the Evap Unit controllers decision. If the Evap Unit controller decides to turn OFF the evaporative cooler, then the Evap Unit and the fan are OFF. So, in the existing implementation, the fan and the Evap Unit do not run unless the controller decides to turn ON the Evap cooler (Evap Unit). Therefore, the exiting implementation is "some sort" of cycling fan (cycling unit) but with FULL fan flow for ZoneTemperatureDeadbandOnOffCycling and ZoneCoolingLoadOnOffCycling Unit Control Methods. ZoneCoolingLoadVariableSpeedFan Unit Control Method modulates the fan flow as the name implies.

What this fix allowed is to modulate the fan flow rate for the ZoneTemperatureDeadbandOnOffCycling and ZoneCoolingLoadOnOffCycling Unit Control Methods to avoid over cooling. This fix makes Evap Unit control similar to how we cycle other zone equipment depending on fan and load.

Now the question is do we want to allow "Continuous Fan Flow" just like the other  zone equipment for Evap Cooler Unit? This will require to turn OFF the evaporative-cooler but allow FULL flow outside air to the zone. This change would be quite different from what the user requested.

Reply:
@Nigusse Ohhhhh - I hadn't realized this unit has a "Cooler Unit Control Method" field. Maybe that simplifies things. 

So, now, let's not add new schedule field. It probably doesn't make sense for a unit like this - who would want a continuous flow of air that gets hot when the water cycles off?

So, it seems the "Cooler Unit Control Method" should dictate the operation, regardless of fan type. If the fan settings are not compatible with the selected control type, then either severe/fatal out, or throw a warning and flip the control choice?

Reply:
@mjwitte Thanks. Will look into "Cooler Unit Control Method" and fan type combination.

Reply:
Existing code of the zone Evap Cooler Model supports five fan types:  "Fan:OnOff", "Fan:SystemModel", "Fan:ComponentModel", "Fan:ConstantVolume" and "Fan:VariableVolume".

If the Cooler Unit Control Method is ZoneCoolingLoadVariableSpeedFan, "Fan:OnOff" and "Fan:ConstantVolume" currently are not allowed and EPlus fatal out with severe error message. 

However, with the other two Cooler Unit Control Methods (ZoneCoolingLoadOnOffCycling and ZoneTemperatureDeadBandOnOffCycling) existing code allows all the five fan types.

Below is the proposed change for ZoneCoolingLoadOnOffCycling and ZoneTemperatureDeadBandOnOffCycling Cooler Unit Control Methods:
(1) fan types allowed will be restricted to "Fan:OnOff" and "Fan:SystemModel"
(2) fatal out with severe error message for the other three fan types
(3) Update existing example files based on the first two items of proposed changes

Reply:
I see only two example files that need to change the fan type in multiple places for the two cooler unit control methods. I am planning to replace the :OnOff, :ConstantVolume, and :VariableVolume fan types with :SystemModel used with the two control methods only. @mjwitte Let me know if you have any objection.

Reply:
@Nigusse Why make the fan types more restrictive now? If that's necessary, then we'll need transition rules to change the fan types.
Also, if it's made more restrictive, Fan:ComponentModel should also be valid.

Reply:
@mjwitte I am just trying to allowing fan types I thought appropriate for these two controller types. Otherwise, I do not have any specific reason. Are you suggesting that we allow all the five fan types for the ZoneCoolingLoadOnOffCycling and ZoneTemperatureDeadBandOnOffCycling Cooler Unit Control Methods as it is done in the existing code?

Reply:
Yes, as long as they produce reasonable results (i.e. power consumption that makes sense).

Reply:
OK. I will revise the code accordingly and push to remote. And I will review the regression results.

Reply:
@mjwitte I created a single zone model to investigate electric power consumption. see below the fan and evap unit electric power for "EvaporativeCooler:Direct:CelDekPad". The Unit's electric power for this evap cooler is the electric power of the Water Pump only.  The supply fan power reported is consistent with the operation mode of the fan type (cycling for Fan:OnOff and Full Flow for Fan:ConstantVolume) when the evap unit is ON depending the controller action. The control type tested is ZoneCoolingLoadOnOffCycling. I see a problem with the electric power (water Pump Electric Power), it shouldn't have been constant for cycling fan/Unit (the first graph) when the PLR is less than 1.0. This should be fixed, but I am afraid this may be the case for the other evap cooler types as well.

![CyclingUnit_FullFlowUnit](https://user-images.githubusercontent.com/5805126/59300373-08dc1900-8c5d-11e9-95f6-6fa99bfd0213.png)

Reply:
But I'm trying to get away from the fan type affecting operation. Why aren't these two cases the same?

Reply:
Do we allow Fan:ConstantVolume fan to cycle? The original code allowed FULL flow regardless of the fan type when the unit ON and causing zone over-cool. Now, this fix allowed the unit/fan to cycle but only with these three fan types (Fan:OnOff, Fan:SystemModel, Fan:ComponentModel).  I can change the fix to allow the unit to cycle regardless of the fan type, if you agree?

Reply:
That's what I thought we agreed on. That's assuming Fan:ConstantVolume will work that way. The IDD notes for it imply otherwise:
```
Fan:ConstantVolume,
       \memo Constant volume fan that is intended to operate continuously based on a time schedule.
       \memo This fan will not cycle on and off based on cooling/heating load or other control
       \memo signals.
```
If that's true, then we should throw an error if it's used with a cycling control option and add a transition rule to replace it with Fan:OnOff when use with the evap cooler.
@Myoldmopar It's time we dump all of these old fans. They should all be marked obsolete for this release. And possibly transitioned away in this release and the next.